root/hist/hist/TFormulaGradientBenchmarks.cxx

```diff
@@ -0,0 +1,87 @@
+#include <TFormula.h>
+#include <TF1.h>
+#include <TFitResult.h>
+#include <TH1.h>
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include <benchmark/benchmark.h>
+
+static void BM_TFormula_GausTFormulaPerf_Clad(benchmark::State &state) {
+  for (auto _ : state) {
+    state.PauseTiming();
+    auto h = new TF1("f1", "gaus");
+    double p[] = {3, 1, 2};
+    h->SetParameters(p);
+    double x[] = {0};
+    TFormula::GradientStorage result_clad(3);
+    TFormula *f = h->GetFormula();
+    state.ResumeTiming();
```

Comment:
Could you move all this initialization out of the loop. Our hot function we want to test is `GradientPar`.